app/components/login/NewUserRegistration.js

```diff
@@ -0,0 +1,123 @@
+import Button from '@material-ui/core/Button';
+import { useStyles } from '../index';
+
+function NewUserRegistration(props) {
+  const [loginError, setError] = useState(false);
+  const user = useSelector(state => state.user);
+  const [initLoad, setInit] = useState(true);
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const dispatch = useDispatch();
+
+  const classes = useStyles();
+
+  const handleChange = (event) => {
+    switch (event.target.id) {
+      case "firstname":
+        setFirstName(event.target.value);
+        break;
+      case "lastname":
+        setLastName(event.target.value);
+        break;
+        case "email":
+        setEmail(event.target.value);
+        break;
+        case "password":
+        setPassword(event.target.value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = () => {
+    setInit(false);
+    setError(false);
+    dispatch(updateUser(user.id, { firstName, lastName, email, password, userId: user.id }));
+    setEmail('');
+    setPassword('');
+    props.history.push(`/`)
+  }
+
+  useEffect(() => {
+    if (user && user.isRegistered) {
+      props.history.push('/');
+    }
+    if (!initLoad && !user.isRegistered) {
+      setError(true);
+    }
+    if (email || password) {
+      setError(false);
+    }
+  });
+
+  return (
+    <div id="loginContainer" className={[classes.root, classes.centerHero].join(' ')}>
+
+      {
+        user.isRegistered &&
+        <div>
+          LoggedIn!
+                    </div>
+      }
+      <form className={classes.root}>
+        <h1 className={classes.header}>Sign up</h1>
+        <h3 className={classes.header}>glad to see you!</h3>
+        {loginError &&
+          <ErrorBar title={"Login failed"} message={"Incorrect email or password"} />
+        }
+        <TextField
+          id="firstname"
+          label="First Name"
+          variant="outlined"
+          value={firstName}
+          onChange={handleChange}
+          className={classes.formElem}
+        />
+        <TextField
+          id="lastname"
+          label="Last Name"
+          variant="outlined"
+          value={lastName}
+          onChange={handleChange}
+          className={classes.formElem}
+        />
+        <TextField
+          id="email"
+          label="Email"
+          variant="outlined"
+          value={email}
+          onChange={handleChange}
+          className={classes.formElem}
+        />
+        <TextField
+          id="password"
+          label="Password"
+          variant="outlined"
+          type="password"
+          value={password}
+          onChange={handleChange}
+          className={classes.formElem}
+        />
+        <Button className={classes.formElem} variant="contained" color="primary" onClick={handleSubmit}>Log In</Button>
```

Comment:
The copy here should be 'Create Account' not 'Log In'
Be careful if you're just copying/pasting huge blocks of code.

Reply:
Done. :) oversight error at 3 am coding side effect.